Feature/choice model #59

A1-frkweiichi opened the pull request:
※コントローラー不要で、モデルを追加しました。
クイズに選択肢を追加しました

- 選択肢はクイズに紐づく
- 選択肢は2つ以上、必須
- 選択肢は1つ以上、正解が必須
上記のバリデーションを実装しています。

A1-frkweiichi committed:
ゲストログイン機能をmergeします

Merge remote-tracking branch 'refs/remotes/origin/develop' into develop

A1-frkweiichi committed:
グーグルアナリティクスのコードをmergeします

Merge remote-tracking branch 'refs/remotes/origin/develop' into develop

A1-frkweiichi committed:
クイズに選択肢を追加しました

- 選択肢はクイズに紐づく
- 選択肢は2つ以上、必須
- 選択肢は1つ以上、正解が必須
上記のバリデーションを実装しています。

ayakuma309 commented:
確認しました！ありがとうございます！
 LGTMです！

ayakuma309 commented:
quiz/showの方は
```
<%= quiz.title %>
<%= quiz.content %>
<%= quiz.image %>
```
しかないので選択肢も付け足せばいいのかなと思いました。

A1-frkweiichi committed:
app/views/quizzes/---.html.erbファイル　i18n日本語化　実装

対象ファイル
app/views/quizzes/_form.html.erb
app/views/quizzes/_quiz.html.erb
app/views/quizzes/edit.html.erb
app/views/quizzes/index.html.erb
app/views/quizzes/new.html.erb
app/views/quizzes/show.html.erb

A1-frkweiichi committed:
i18n 日本語化　実装

A1-frkweiichi committed:
QuizにChoiceを追加する

- 選択肢は4つまで、2つ以上登録必須
- 正解も4つまで、1つ以上登録必須

furukawaeiichi commented:
下記の確認お願いします。

選択肢の実装
- 選択肢は4つまで、2つ以上登録必須
- 正解も4つまで、1つ以上登録必須

クイズ、選択肢
- i18n適応

A1-frkweiichi committed:
編集時のみ、レコードを削除できるチェックボックスを表示する

A1-frkweiichi committed:
Quizのmodel, controllerをデバッグ

編集時も、選択肢が2未満になることを防ぎました。
また、スリムになるように共通メソッドを作り直しました。
def build_choices
  unbuilt_choices = [4 - @quiz.choices.size, 0].max
  unbuilt_choices.times { @quiz.choices.build }
end

furukawaeiichi commented:
- 編集時のみ、レドーどを削除できるチェックボックスが表示されるようにしました。

- 編集時も、選択肢が2未満にならないよう制限しました。
また、スリムになるように共通メソッドを作り直しました。
def build_choices
  unbuilt_choices = [4 - @quiz.choices.size, 0].max
  unbuilt_choices.times { @quiz.choices.build }
end

A1-frkweiichi committed:
Merge branch 'develop' into feature/choice_controller_#59
